kernel/ramfs.c

```diff
@@ -738,6 +760,9 @@ static int _fs_open(
 
         if ((flags & O_APPEND))
             file->offset = inode->buf.size;
+
+        if (inode->vcallback)
+            ECHECK((*inode->vcallback)(&file->vbuf));
```

Comment:
Why don't we use `inode->buffer` to unify the getter functions?

Reply:
Thread-safety was one reason for keeping the buffer at the file level.

Reply:
This doesn't eliminate race condition still, right?

Reply:
Yes agree :( There is still a race condition, concurrent accesses on the same file within a thread group. @mikbras and I had discussed this but couldn't find a suitable solution.
Also Linux also stores /proc virtual files under `file->private_data`.